examples/access_traits/main.cpp

```diff
@@ -0,0 +1,82 @@
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+struct Data
+{
+  float *d_x;
+  float *d_y;
+  float *d_z;
+  int N;
+};
+
+struct Data2
+{
+  float *d_x;
+  float *d_y;
+  float *d_z;
+  float *d_r;
+  int N;
+};
+
+namespace ArborX
+{
+namespace Details
+{
+namespace Traits
+{
+template <>
+struct Access<Data>
+{
+  __device__ inline static Point get(Data const &d, int i)
+  {
+    return {{d.d_x[i], d.d_y[i], d.d_z[i]}};
+  }
+  inline static typename std::size_t size(Data const &d) { return d.N; }
+  using Tag = PointTag;
```

Comment:
Would this tag dictate the return type of `get()`? i.e. `Tag` dispatches an implementation of the BVH which calls `get()` with an expected return type.